manager/assets/modext/widgets/core/modx.combo.js

```diff
@@ -783,121 +811,243 @@ Ext.reg('modx-combo-property-set',MODx.combo.PropertySet);
+        }
 
-    ,handleChangeParent: function(node,e) {
-        var t = Ext.getCmp('modx-resource-tree');
-        if (!t) { return false; }
-        t.disableHref = true;
+        var id = node.id.split('_'),
+            id = id[1];
 
-        var id = node.id.split('_'); id = id[1];
         if (id == this.config.currentid) {
-            MODx.msg.alert('',_('resource_err_own_parent'));
+            MODx.msg.alert(_('warning'), _('resource_err_own_parent'));
+            
             return false;
         }
 
-        var ctxf = Ext.getCmp(this.config.contextcmp);
-        if (ctxf) {
-            var ctxv = ctxf.getValue();
-            if (node.attributes && node.attributes.ctx != ctxv) {
-                ctxf.setValue(node.attributes.ctx);
+        var contextField = Ext.getCmp(this.config.contextcmp);
+        
+        if (contextField) {
+            if (node.attributes && node.attributes.ctx !== contextField.getValue()) {
+                contextField.setValue(node.attributes.ctx);
             }
         }
-        this.fireEvent('end',{
-            v: node.attributes.type != 'modContext' ? id : node.attributes.pk
-            ,d: Ext.util.Format.stripTags(node.text)
+
+        this.fireEvent('setParentValue', {
+            value   : node.attributes.type != 'modContext' ? id : node.attributes.pk,
+            label   : Ext.util.Format.stripTags(node.text)
         });
+
         e.preventDefault();
         e.stopEvent();
+        
         return true;
+    },
+    setParentValue: function(value) {
+        this.currentValue = false;
+        
+        this.setValue(value.label || value.value);
+
+        var parentField = Ext.getCmp(this.config.parentcmp);
+
+        if (parentField) {
+            parentField.setValue(value.id);
+        }
+        
+        var formPanel = Ext.getCmp(this.config.formpanel);
+        
+        if (formPanel) {
+            formPanel.fireEvent('fieldChange');
+        }
+
+        this.enableResourceTree();
     }
 });
+
 Ext.reg('modx-field-parent-change',MODx.ChangeParentField);
 
+MODx.combo.Resource = function(config) {
+    config = config || {};
+
+    Ext.applyIf(config,{
+        url         : MODx.config.connector_url,
+        baseParams  : {
+            action      : 'resource/getlist',
+            combo       : 1,
+            limit       : 0,
+            ignore      : config.currentid || ''
+        },
+        fields      : ['id', 'pagetitle', 'longtitle', 'context_key', 'context_name', 'time'],
+        name        : 'resource',
+        hiddenName  : 'resource',
+        displayField : 'pagetitle',
+        valueField  : 'id',
+        pageSize    : 20,
+        allowBlank  : true,
+        editable    : true,
+        typeAhead   : true,
+        tpl         : new Ext.XTemplate('<tpl for=".">' +
+            '<tpl if="!Ext.isEmpty(this.getGroup(values.context_name, values.time))">' +
+                '<div class="x-combo-list-group">{this.label}</div>' + 
```

Comment:
Same as earlier, please add `htmlEncode` to `this.label` to prevent potential XSS.